chore: upgrade lerna (#2836)
open-telemetry/opentelemetry-js-contrib: CONTRIBUTING.md

The `opentelemetry-js-contrib` project is written in TypeScript.

As a general rule, installing from the root directory should always be done first before anything else.
Packages within this repository might have dependencies between them. This means the dependencies should
be built before if you want to `compile` or `test` the changes you've made in a package. Each package
has a script to ensure these dependecies are ready.

The required steps to start development on a pacakge are:

- `npm ci` from root folder to install dependencies ([see npm-ci docs](https://docs.npmjs.com/cli/v10/commands/npm-ci))
- `cd` into the pacakge you want to apply changes.

> - `npm run setup:dev` checks for dependencies of the current within the repository and prepares them.

Reviewer: ```suggestion
- `npm run setup:dev` compiles the TypeScript files for this package and its dependencies within the repository.
```

Each of these commands can also be run in individual packages, as long as the initial install and compile are done first in the root directory.

> If you're going to work on a single package and want to quickly jump into its development you can make use of the `focus` script. This scritp will run

Reviewer 2: ```suggestion
If you're going to work on a single package and want to quickly jump into its development you can make use of the `focus` script. This script will run
```

> the necessary tasks to have only that package ready for development saving time. For example, if you want to work with `@opentelemetry/resource-detector-aws`

Reviewer 2: nit: 
```suggestion
the necessary tasks to have only that package ready for development, saving time. For example, if you want to work with `@opentelemetry/resource-detector-aws`
```

I'm not entirely sure about that comma - I'm not a native speaker and commas in the English language are a mystery to me 😅

run the following command in the root folder

```sh
npm run focus @opentelemetry/resource-detector-aws

> opentelemetry-contrib@0.1.0 focus
> nx run-many -t compile -p @opentelemetry/resource-detector-aws

    ✔  nx run @opentelemetry/instrumentation-fs:version:update (796ms)
    ✔  nx run @opentelemetry/contrib-test-utils:compile (3s)
    ✔  nx run @opentelemetry/instrumentation-fs:compile (3s)
    ✔  nx run @opentelemetry/resource-detector-aws:compile (3s)

 ———————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————

 >  NX   Successfully ran target compile for project @opentelemetry/resource-detector-aws and 3 tasks it depends on (7s)
```

> Once the command is done you can `cd` into the package and start using the usual commands.

Reviewer: (non-blocking)  I'm not opposed to this command, but it would be kind of nice to not have to type out both (a) the package name and (b) the directory for that package.

Possible alternative:

```
cd detectors/node/opentelemetry-resource-detector-aws   # or whatever package to focus on
npm run compile:include-deps
```

Or perhaps others would find a command at the top-level more convenient? **My personal** typical development process is that I `cd` into the package dir I am working on and work from there. So an `npm run ...` script **in that package dir** would be convenient for *me*.

Author: Since I was already removing scripts from each `package.json` I didn't want to add anything else. But I think its a fair point. At the end you have to `cd` into that directory to run the scripts there so it would be more convenient.

Checklist:
- [ ] remove `focus` script
- [ ] add `compile:include-deps` script in each package
- [ ] update `contributing.md`: remove focus section
- [ ] update `contributing.md`: rewrite [compile at the top requirement](https://github.com/open-telemetry/opentelemetry-js-contrib/blob/main/CONTRIBUTING.md?plain=1#L111-L113)
